File: data/llama3_8B_ins_results/readable_format_HPEC/cpp/file275.cpp
#include <iostream>
#include <omp.h>

int main() {
    int var = 0;
    int i;

    #pragma omp target teams distribute parallel do map(tofrom:var) device(0)
    for (i = 1; i <= 100; i++) {
        #pragma omp critical(addlock)
        var++;
        #pragma omp end critical(addlock)
    }
    #pragma omp end teams distribute parallel do
    #pragma omp end target

    std::cout << var << std::endl;
    return 0;
}

File: data/StarCoder2/readable_format_HPEC/fortran/file275.f90
program DRB148_critical1_orig_gpu_yes
    use omp_lib
    implicit none

    integer :: var, i
    var = 0

    !$omp target map(tofrom:var) device(0)
    !$omp teams distribute parallel do
    do i = 1, 100
        !$omp critical(addlock)
        var = var+1
        !$omp end critical(addlock)
    end do
    !$omp end teams distribute parallel do
    !$omp end target

    print*,var
end program
